py/server/deephaven/table.py

```diff
@@ -518,21 +519,98 @@ def iter_rows(self, cols: Optional[Union[str, Sequence[str]]] = None, *, chunk_s
+        context will be released after the generator is destroyed. That can happen implicitly when the generator
+        is used in a for-loop. When the generator is not used in a for-loop, to prevent resource leaks, it must be closed
+        after use by either (1) by setting it to None, (2) using the del statement, or (3) calling the close() method on it.
+
+        Args:
+            cols (Optional[Union[str, Sequence[str]]]): The columns to read. If None, all columns are read. Default is None.
+            name (str): The name of the named tuple. Default is 'Deephaven'.
+
+        Returns:
+            A generator that yields a named tuple or regular tuple for each row in the table
+        """
+        from deephaven._table_reader import _table_reader_rows # to prevent circular import
+        named_tuple_class = namedtuple(name, cols or [col.name for col in self.columns], rename=False)
+
+        for row in _table_reader_rows(self, cols):
+            yield named_tuple_class(**row)
+
+    def iter_chunk_dict(self, cols: Optional[Union[str, Sequence[str]]] = None, *, chunk_size: Optional[int] = 4096)-> Generator[Dict[str, np.ndarray], None, None]:
+        """ Returns a generator that reads one chunk of rows at a time from the table into a dictionary. The dictionary
+        is a map of column names to numpy arrays of the column data type.
+
+        If the table is refreshing and no update graph locks are currently being held, the generator will try to acquire
+        the shared lock of the update graph before reading the table data. This provides a consistent view of the data.
+        The side effect of this is that the table will not be able to refresh while the table is being iterated on.
+        Additionally, the generator internally maintains a fill context. The auto acquired shared lock and the fill
+        context will be released after the generator is destroyed. That can happen implicitly when the generator
+        is used in a for-loop. When the generator is not used in a for-loop, to prevent resource leaks, it must be closed
+        after use by either (1) by setting it to None, (2) using the del statement, or (3) calling the close() method on it.
+
+        Args:
+            cols (Optional[Union[str, Sequence[str]]]): The columns to read. If None, all columns are read.
+            chunk_size (int, Optional): The number of rows to read at a time. Default is 4096.
+
+        Returns:
+            A generator that yields a dictionary of column names to numpy arrays.
+
+        Raises
+            ValueError
+        """
+        from deephaven._table_reader import _table_reader_chunks  # to prevent circular import
+
+        return _table_reader_chunks(self, cols=cols, row_set=self.j_table.getRowSet(), chunk_size=chunk_size,
                                         prev=False, to_numpy=True)
 
+    def iter_chunk_tuple(self, cols: Optional[Union[str, Sequence[str]]] = None, *, chunk_size: Optional[int] = 4096,
```

Comment:
This return type doesn't make sense to me; we are saying it's a 2-element tuple return, whereas I expect a tuple that has size equal to the number of columns. I think the appropriate tuple type is `tuple[Any, ...]`. (Maybe you meant to have some sort of `NamedTuple` type?)

Reply:
Yeah, missed the change after copy-paste.